src/main/scala/cognite/spark/v1/SdkV1Relation.scala

```diff
@@ -88,6 +90,17 @@ abstract class SdkV1Relation[A <: Product, I](config: RelationConfig, shortName:
       val rowOfAllFields = toRow(item)
       Row.fromSeq(indicesOfRequiredFields.map(idx => rowOfAllFields.get(idx)))
     }
+
+  // scalastyle:off no.whitespace.after.left.bracket
+  def updateByExternalIdHandler[
+      T <: WithSetExternalId,
+      R <: WithExternalId,
+      U <: UpdateByExternalId[A, T, IO]](updates: Seq[R], resource: U)(
```

Comment:
I'm confused by the names :)
`R` should be the "read" type, for example `Asset`, returned from listing.
`U` should be the "update" type, for example `AssetUpdate`.
`C` (which we don't have here) is the "create" type, `AssetCreate`.

Can we find better names? May even have to use longer names to make it clear, even if that can also be confusing, for example `Resource <: UpdateByExternalId[...]`. However, couldn't we just do `resource: UpdateByExternalId[...]`?

Reply:
I tried doing `resource: UpdateByExternalId[...]` without having U already as a parametrization of the method, but it was not happy with that. I am not sure if it is possible to define a generic in the parameter list?

Reply:
I have changed it to be that U is xyUpdate, C is xyCreate or xyUpsertSchema, and T is the resource type